### PR TITLE
DAQ related bug fixes + improvement of the TrigResultsNavigator class + new Menus for onSpill,offSpill,extrPos

### PR DESCRIPTION
**Pull request by gianipez**
- some bug fixes in a module and config of the DAQ package
- TriggerResultsNavig now gets the list of the paths in the trigger_paths also in a job where the triggers are not run 
- added menu files created at compilation level for onSpill, offSpill, extrPos
- added UeP and UeM CosmicHelix trigger paths (included in the offSpill Menu)
- converted the CosmicSeedDeM/P to CosmicHelixDeM; the new sequences stop at the pattern_reco stage (no KalSeedFit anymore)
- added the StraightTrack trigger path in the extrPos, offSpill and allPaths menus

### CODE REVIEW

**Comment by gianipez** [comment]
Hi folks, I didn't apply yet the change for setting the trigger_bits in the Menu files. I believe these files will be mostly used by myself and people running MC sim+reco together with the trigger reco. Incorporating the trigger_paths with a bit set with an additional one for the MC+reco path is possible, but we need to agree to a convention. For example, if we agree to run the MC production in two separate stages: 
1. MC sim + Offline reco
2. Trigger & ntupling

In this way, in the Menu files, I can set my trigger bits from 0 to N-1 without any problem. This solution although would impose us to use more disk storage I guess.

 In principle, I can also set the bits from 1 to N, and leave the first one "free", BUT I can imagine scenarios where maybe we want to run different paths with different MC selections...

File: scripts/build/python/sconstruct_helper.py
Comment (kutschke) [inline scripts/build/python/sconstruct_helper.py:168]: Why do we need to build CRVResponse for code to be shipped to the trigger.  The whole point of the notNeeded list is that we do not want the trigger builds to contain MC code and CRVResponse is a MC package.  Please let me know what you need from that package and we can discuss refactoring what lives in which directory.

**Review by kutschke — CHANGES_REQUESTED** [review]
My apologies for the delay in getting to this. 

Please send a list of the names of all files that you modified in a text file so that I can manipulate the names programatically.  I want to get a systematic view of what's been touched.

Also, please suggest people who can review this PR.

File: scripts/build/python/sconstruct_helper.py
Comment (gianipez) [reply scripts/build/python/sconstruct_helper.py:168]: Hi Rob, CRVRsponse house MC and Reco code, that's why I need it. I agree that it would be good if someone from the CRV group would implement the separation. I remember I put this comment in a MR a long time ago, but maybe it got ignored.

Comment (kutschke) [reply scripts/build/python/sconstruct_helper.py:168]: Thanks Giani.  I will speak with Ralf and cc you - I agree that he should do the work.

File: Trigger/src/ReadTriggerInfo_module.cc
Comment (kutschke) [inline Trigger/src/ReadTriggerInfo_module.cc:1306]: Please use an enum rather than than hard coded numbers.   Ask Dave if he supports an enum within BTrk and if it will be in the future KinKal.   If not, it is OK to use PDGCode::e_minus.  We moved PDGCode.hh out of MCDataProducts and into DataProducts so that it was no longer considered "MC" and could be used with data.  You should do this everywhere in the code base.

File: Trigger/src/ReadTriggerInfo_module.cc
Comment (kutschke) [inline Trigger/src/ReadTriggerInfo_module.cc:86]: When you change these numbers, do you need to make the corresponding changes in other parts of the code or in fcl or config files?  If so, what would it take to get a single point of maintenance?

File: TrkHitReco/src/StrawHitReco_module.cc
Comment (kutschke) [inline TrkHitReco/src/StrawHitReco_module.cc:55]: If a parameter is normally provided by fcl, we do not want default values in the code.   Same comment in many places.

File: DAQ/src/StrawAndCaloDigisFromFragments_module.cc
Comment (kutschke) [inline DAQ/src/StrawAndCaloDigisFromFragments_module.cc:466]: Does the copy does an implicit reserve or do we need to do that ourselves?  I bet it does but I don't know.  Please check.   I agree that you don't need to default construct everything in the vector.

**Comment by kutschke** [comment]
@FNALbuild run build tests

File: TrkHitReco/src/StrawHitReco_module.cc
Comment (gianipez) [reply TrkHitReco/src/StrawHitReco_module.cc:55]: I'm ok in removing it. I only changed the value of the default previously set by @bonventre

Comment (kutschke) [reply TrkHitReco/src/StrawHitReco_module.cc:55]: Please also check that the relevant fcl files do set it.

File: DAQ/src/StrawAndCaloDigisFromFragments_module.cc
Comment (gianipez) [reply DAQ/src/StrawAndCaloDigisFromFragments_module.cc:466]: Thanks for checking this. `std::copy` doesn't call any reserve internally. I'll add it now.

File: Trigger/src/ReadTriggerInfo_module.cc
Comment (gianipez) [reply Trigger/src/ReadTriggerInfo_module.cc:86]: These numbers cannot be part of the list of data-members of the class, because they define the size of the arrays of histogram I set in my `struct` instances. I usually change it when the Menu I test gets longer, which happens not that frequently.

File: Trigger/src/ReadTriggerInfo_module.cc
Comment (kutschke) [reply Trigger/src/ReadTriggerInfo_module.cc:1306]: I will also leave this to Dave to decide whether to do it now or to defer for a later PR

Comment (brownd1978) [reply Trigger/src/ReadTriggerInfo_module.cc:1306]: Synchronizing pdgcodes across Offline, BTrk and KinKal is a bigger issue.  Fortunately they all use the same values.  It's clear though this code should use PDGCode values, not hardcoded numbers.

File: TrkHitReco/src/StrawHitReco_module.cc
Comment (brownd1978) [reply TrkHitReco/src/StrawHitReco_module.cc]: Lets defer any changes here.  We can create an issue that the default values should be re-evaluated.

**Review by brownd1978 — CHANGES_REQUESTED** [review]
This should be good to go after fixing the PDGCodes issue.

File: Trigger/src/ReadTriggerInfo_module.cc
Comment (kutschke) [reply Trigger/src/ReadTriggerInfo_module.cc:1306]: I added the issue #292 and will mark it resolved.

File: TrkHitReco/src/StrawHitReco_module.cc
Comment (kutschke) [reply TrkHitReco/src/StrawHitReco_module.cc]: Created issue #293

**Review by kutschke — APPROVED** [review]
I have made issues for the outstanding comments and will approve.

File: Trigger/src/ReadTriggerInfo_module.cc
Comment (gianipez) [reply Trigger/src/ReadTriggerInfo_module.cc]: OK, no problem. Done.

**Comment by gianipez** [comment]
@FNALbuild run build tests

**Comment by gianipez** [comment]
Hi folks,

Thanks again for the comments. I integrated the required changes. 

Cheers,

Giani

**Review by brownd1978 — APPROVED** [review]

**Comment by brownd1978** [comment]
Thanks Giani.  I want to check a few things about code merged last night, if that comes out as expected I will merge this today.

**Pull request merged**
